src/arithmetic.jl

```diff
@@ -19,7 +19,7 @@ function plus_rev(a::Interval, b::Interval, c::Interval)  # a = b + c
     return a, b_new, c_new
 end
 
-plus_rev(a,b,c) = plus_rev(promote(a,b,c)...)
+plus_rev(a::F, b::F, c) where {T, F<:Interval{T}} = plus_rev(a, b, F(c))
```

Comment:
shouldn't this (and the others) be

```julia
function plus_rev(a::Interval{T}, b::Interval{S}, c::Interval(R})
  TSR = promote_type(T, S, R) # not sure if you can do it in one line, but you get the idea
  F = Interval{TSR}
  return plus_rev(F(a), F(b), F(c)
```

but it gets pretty tedious... I get the idea of not having promotion Float64 --> Interval{Float64}, but I think a promotion rule for `Interval{T}, Interval{S}` would be harmless.

In both cases something like `plus_rev(1.0, 1..2, 3..4)` wouldn't work. This can be a bug (if you think in the more ducktyping way) or a feature (if you come from a strong static typing programming language)

Reply:
Regarding the second method, which previously had a promotion, I truly don't recall what was the motivation. I guess it was a way of avoiding using `promote`... 

In any case, let me note that many operations in the 1.0-dev branch of IA impose the same `Interval{T}` for the operands: e.g., `(1.0f0 .. 2.f0) - ( 1.5 .. 2.5)` throws a `MethodError` due to ambiguities...

Reply:
if it's because of ambiguity, then it's a bug (or at least misleading error message). In general, it's not necessarily wrong to disallow implicit conversions, strongly typed languages do it and for good reasons (see [this accident](https://en.wikipedia.org/wiki/Ariane_flight_V88) )

Reply:
but this is probably tangential to this PR